lib/events/dynamoevents/dynamoevents_test.go

```diff
@@ -437,15 +437,15 @@ func TestConfig_CheckAndSetDefaults(t *testing.T) {
 }
 
 // TestEmitSessionEventsSameIndex given events that share the same session ID
-// and index, the emit should fail, avoiding any event to get overwritten.
+// and index, the emit should succeed.
 func TestEmitSessionEventsSameIndex(t *testing.T) {
 	ctx := context.Background()
 	tt := setupDynamoContext(t)
 	sessionID := session.NewID()
 
 	require.NoError(t, tt.log.EmitAuditEvent(ctx, generateEvent(sessionID, 0)))
 	require.NoError(t, tt.log.EmitAuditEvent(ctx, generateEvent(sessionID, 1)))
-	require.Error(t, tt.log.EmitAuditEvent(ctx, generateEvent(sessionID, 1)))
+	require.NoError(t, tt.log.EmitAuditEvent(ctx, generateEvent(sessionID, 1)))
```

Comment:
Hmm, do we want the audit log to be able to be overwritten.

I feel like the original behavior of "first write wins" is more correct for an audit log.

Should we instead just not treat already exists as an error?

Reply:
> Hmm, do we want the audit log to be able to be overwritten.

This will only happen if the `disable_conflict_check` is provided. Otherwise, the event will always need a unique session_id/event_index pair. The fallback logic only sets the event index to a different value and tries to put the item again. If there is still a conflict (very unlikely to happen since we're using unix nano and this is attached to the session), the event emission will still fail.

Reply:
If it makes it clearer, we can add another test that causes the fallback to also fail. In this case, the `EmitAuditEvent` will return an error.